src/client/tile.cpp

```diff
@@ -976,8 +983,17 @@ video::ITexture* TextureSource::generateTextureFromMesh(
 	return rtt;
 }
 
-video::IImage* TextureSource::generateImage(const std::string &name)
+video::IImage* TextureSource::generateImage(const std::string &name,
+		normal_presence is_normalmap)
 {
+	// Search for _normal.png if its presence isn't known yet
+	if (is_normalmap == NORMAL_UNKNOWN) {
+		is_normalmap = (name.find("_normal.png") != std::string::npos)
```

Comment:
What's the advantage of this argument-passing method compared with a `bool is_normalmap = (name.find("_normal.png") != std::string::npos);` for each function call? `find` is quite fast, if you're doing this for performance reasons.

Reply:
Searching for _normal.png needs to be done as early as possible because generateImage is called recursively.
is_normalmap should stay the same value for a texture, so searching should be performed only one time for each texture.
I've changed the comment to explain this behaviour.